src/com/nutrons/framework/Robot.java

```diff
@@ -69,6 +71,10 @@ protected final void robotInit() {
    */
   @Override
   public final void robotMain() {
+    Command auto = this.registerAuto();
+    if (auto != null) {
+      this.competitionStream().filter(x -> x == AUTO).subscribe(x -> auto.terminable(competitionStream().filter(y -> y != AUTO)).execute());
```

Comment:
Might want to move `auto.terminable(...)` part to go on the `this.registerAuto()` line. Would be easier to read.